src/main/java/com/github/mproberts/rxtools/map/SubjectMap.java

```diff
@@ -331,8 +331,68 @@ public void runIfBound(K key, final Runnable runnable)
         }
 
         if (isBound) {
-            runnable.run();
+            return processNewFaultForKey(key);
         }
+        return Completable.complete();
+    }
+
+    private Completable processNewFaultForKey(final K key)
+    {
+        return Completable.defer(new Callable<CompletableSource>() {
+            @Override
+            public CompletableSource call() throws Exception {
+                return new CompletableSource() {
+                    @Override
+                    public void subscribe(CompletableObserver completableObserver) {
+                        Function<K, Single<V>> faultHandler = _faultHandler;
+
+                        emitFault(key);
+
+                        if (faultHandler != null) {
+                            try {
+                                Single<V> fault = faultHandler.apply(key);
+
+                                fault.doOnSuccess(new Consumer<V>() {
+                                    @Override
+                                    public void accept(V v) throws Exception {
+                                        WeakReference<Processor<V, V>> weakSource = _weakSources.get(key);
+                                        Processor<V, V> processor;
+                                        if (weakSource != null && (processor = weakSource.get()) != null ) {
+                                            processor.onNext(v);
+                                        }
+                                    }
+                                }).toCompletable().subscribe(completableObserver);
+                            } catch (Exception e) {
+                                completableObserver.onError(e);
+                            }
+                        } else {
+                            completableObserver.onComplete();
```

Comment:
`onError()` sounds more appropriate

Reply:
actually it doesn't, if you subscribe to faults you don't have a fault handler, so maybe this should still succeed?

Reply:
Yeah we shouldn't error people for not having fault handler setup, it just means we wouldn't need to process the fault since nobody is available to process it

Reply:
I mean maybe we should, but there's the old way to setup fault handlers so this would force them to use the new handler if using this function